src/MoBi.Presentation/MenusAndBars/ContextMenus/ContextMenuForObservedDataFolder.cs

```diff
@@ -41,6 +50,15 @@ public override IEnumerable<IMenuBarItem> AllMenuItems()
          if (groupMenu.AllItems().Any())
             yield return groupMenu;
 
+         if (_treeNode.HasChildren)
+         {
+            var colorGroupingButton = CreateMenuCheckButton.WithCaption(Captions.ColorGroupObservedDataContextMenu);
+            colorGroupingButton.Checked = _userSettings.ColorGroupObservedDataFromSameFolder;
```

Comment:
use the WithChecked and WithCheckedAction extensions methods

Reply:
done